src/app/ngrx/reducers/selectors/downloads-chart.selectors.ts

```diff
@@ -1,68 +1,73 @@
 import { createSelector } from '@ngrx/store';
 import { TimeseriesChartModel } from 'ngx-prx-styleguide';
-import { RouterModel, CHARTTYPE_EPISODES, CHARTTYPE_PODCAST, CHARTTYPE_STACKED } from '../models';
+import { Episode, RouterParams, CHARTTYPE_EPISODES, CHARTTYPE_PODCAST, CHARTTYPE_STACKED } from '../models';
 import { selectRouter } from './router.selectors';
-import { EpisodeModel } from '../episode.reducer';
-import { selectSelectedPageEpisodes } from './episode.selectors';
+import { selectRoutedPageEpisodes } from './episode.selectors';
 import { PodcastMetricsModel } from '../podcast-metrics.reducer';
-import { selectSelectedPodcastMetrics } from './podcast-metrics.selectors';
+import { selectRoutedPodcastMetrics } from './podcast-metrics.selectors';
 import { EpisodeMetricsModel } from '../episode-metrics.reducer';
-import { selectEpisodeMetrics } from './episode-metrics.selectors';
+import { selectRoutedEpisodePageMetrics } from './episode-metrics.selectors';
 import { metricsData } from '../../../shared/util/metrics.util';
 import { mapMetricsToTimeseriesData, subtractTimeseriesDatasets, getTotal,
-  neutralColor, standardColor } from '../../../shared/util/chart.util';
+  neutralColor, standardColor, getColor } from '../../../shared/util/chart.util';
 
 export const selectDownloadChartMetrics = createSelector(
```

Comment:
This selector is composed of all the selectors with data we need to put together the data for the chart. (Sorry if this is too redux 101) whenever any of this state changes, the selector is recomputed and the components react to the change.